pyccl/halos/profiles_2pt.py

```diff
@@ -106,3 +106,72 @@ def fourier_2pt(self, prof, cosmo, k, M, a,
                 raise ValueError("prof2 must be the same as prof")
 
         return prof._fourier_variance(cosmo, k, M, a, mass_def)
+
+
+class Profile2ptR(Profile2pt):
+    """ This class inherits from the 1-halo 2-point correlator between two
+    halo profiles. Sometimes we need not assume that the 1-halo terms are
+    fully correlated. We introduce a term `r_corr` which scales the product
+    of the fourier halo profiles by `(1+r_corr)`. In the trivial case where
+    `r_corr=0` this returns the 2-point correlator of the parent class.
+    Example usecases can be found in arXiv:1909.09102 and arXiv:2102.07701.
+    """
+    def __init__(self, r_corr=0.):
+        self.r_corr = r_corr
+
+    def update_parameters(self, r_corr=None):
+        """ Update any of the parameters associated with this 1-halo
+        2-point correlator. Any parameter set to `None` won't be updated.
+        """
+        if r_corr is not None:
+            self.r_corr = r_corr
+
+    def fourier_2pt(self, prof, cosmo, k, M, a, r_corr=None,
```

Comment:
if `r_corr` is an attribute of the object, why pass it here as a parameter?
I would choose either one or the other

Reply:
Yes, I agree. That's why I personally wouldn't make a separate class for `Profile2ptR`; I would just add `r_corr` as an argument of `Profile2pt.fourier_2pt`.
I don't see why `Profile2pt` prohibits this by:

>     More complicated cases should be implemented by subclassing
>     this class and overloading the :meth:`~Profile2pt.fourier_2pt`
>     method.

Adding `r_corr` in `Profile2ptR` would simplify things.

Reply:
I'm fine with you modifying `Profile2pt` if you prefer!

Reply:
(changing the docstring accordingly)